### DISCUSSION TITLE
adding temperature sensors

### DISCUSSION THREAD

**Discussion by Edhall1044** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#98] [Q&A]
Can I get an example of how to add a temperature sensor? Thanks.

**Comment by BigBobbas** [comment]
Hi there is an example [here](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/instructions/make%20it%20your%20own.md#displaying-sensor-information>) 
If you need any further information, please let us know.
Thanks
Rob

**Comment by youkorr** [comment]
you have to put the sensor in text sensor  

exemple
text sensor:

```
 - platform: homeassistant
    id: liv_temp
    entity_id: sensor.temp_entree_temperature
    internal: true  
```

**Reply by youkorr** [reply]
and the rest of the tutorial is on the page https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/instructions/make%20it%20your%20own.md

**Comment by Edhall1044** [comment]
I've done this and the sensor isn't showing up. At first it wouldn't
compile, after I worked out the spaces, it at least does that
I may try a different one to see if it makes a difference.

**Reply by BigBobbas** [reply]
If you view the logs for the device from the ESPHome dashboard , do you see your temperature sensor appearing with it's current reading?
if you could paste in a message the parts of the config that you have changed and I can take a look.

Thanks
Rob

**Comment by Edhall1044** [comment]
- platform: touchscreen
    page_id: climate_page
    id: climate1
    internal: true
    x_min: 5
    x_max: 105
    y_min: 90
    y_max: 170
    on_click:
      min_length: 10ms
      max_length: 500ms
      then:
        - lambda: |-
            it.printf(40, 120, id(my_font3), white, "%.f",
id(out_hum).state);
        - component.update: s3_box_lcd

error message
Compiling .pioenvs/the-new-jarvis-22/src/main.o
/config/esphome/the-new-jarvis-22.yaml:
In lambda function: /config/esphome/the-new-jarvis-22.yaml:2599:7: error:
'it' was not declared in this scope it.printf(40, 120, id(my_font3), white,
"%.f", id(out_hum).state); ^ /config/esphome/the-new-jarvis-22.yaml:2599:7:
note: suggested alternative: 'int' it.printf(40, 120, id(my_font3), white,
"%.f", id(out_hum).state); ^ int ***
[.pioenvs/the-new-jarvis-22/src/main.o] Error 1

any ideas?

Thanks

**Reply by BigBobbas** [reply]
you have added the display lambda to a binary_sensor (touch screen mapped area) . This needs to be under `display:` in the page section where you want the sensor info to be displayed.

**Comment by fowlerjk** [comment]
I'm having a similar issue.

sensor definition:
```
text_sensor:
  - platform: homeassistant
    id: front_room_lighting_group_state
    entity_id: light.zha_controller_front_room_lighting_group
    internal: true

  - platform: homeassistant
    id: kgmu_temperature
    entity_id: sensor.weather_accuweather_116_brook_dr_realfeel_temperature
    internal: true
```

display definition:
```
          it.printf(220, 75, id(icon_font_80), blue,"\U000F18D7");
          it.printf(240, 120, id(my_font3), white, "%.f", id(kgmu_temperature).state); // outside temp
```

On the home page I get the same temperature as the S3 sensor.
On the screen saver, I get "0"

**Reply by BigBobbas** [reply]
try adding your temperature sensor as a `sensor:` rather than a `text_sensor` , and watch the logs for the device from a terminal or from ESPHome dashboard to confirm the values are correct.

**Reply by fowlerjk** [reply]
When I did that, it would not compile.

```
Linking .pioenvs/esp32-s3box-3/firmware.elf
/data/cache/platformio/packages/toolchain-xtensa-esp32s3/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: .pioenvs/esp32-s3box-3/src/main.cpp.o:(.literal._Z5setupv+0x4d8): undefined reference to `_ZTVN7esphome13homeassistant19HomeassistantSensorE'
/data/cache/platformio/packages/toolchain-xtensa-esp32s3/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: .pioenvs/esp32-s3box-3/src/main.cpp.o:(.literal._Z5setupv+0x4dc): undefined reference to `_ZTVN7esphome13homeassistant19HomeassistantSensorE'
collect2: error: ld returned 1 exit status
*** [.pioenvs/esp32-s3box-3/firmware.elf] Error 1
========================= [FAILED] Took 182.43 seconds =========================
```

**Reply by fowlerjk** [reply]
Wait a minute...just found the option of cleaning the build file. Trying that now.

**Reply by fowlerjk** [reply]
Did that, took forever, but now it does show. Hopefully somebody with the same problem will find this and (1) establish it as a sensor (not a text_sensor) and (2) clean the build file. May I suggest noting this in the howto?
